apps/meteor/app/push/server/apn.ts

```diff
@@ -1,22 +1,46 @@
+import type { IAppsTokens, RequiredField } from '@rocket.chat/core-typings';
 import apn from 'apn';
 import EJSON from 'ejson';
 
+import type { PushOptions, PendingPushNotification } from './definition';
 import { logger } from './logger';
 
-let apnConnection;
-
-export const sendAPN = ({ userToken, notification, _removeToken }) => {
-	if (typeof notification.apn === 'object') {
-		notification = Object.assign({}, notification, notification.apn);
+let apnConnection: apn.Provider | undefined;
+
+// the definition included in the module does not include the set-only attributes
+class ApnNotification extends apn.Notification {
```

Comment:
instead of declaring a new class that needs to be used with a type cast later on, I'd actually recommend augmenting the "official" one:
```ts
declare module 'apn' {
	// eslint-disable-next-line @typescript-eslint/naming-convention
	interface Notification {
		setContentAvailable: (value: boolean | 1 | 0) => void;
		category: string | undefined;
		body: string;
		title: string;
	}
}
```

Reply:
Benefits of taking so long to open a PR, I guess. 
When I first wrote this code this augmentation wouldn't work because those attributes only have setters. Typescript 5.1 added support for different types on setters/getters and now this works.